source/s3_default_meta_request.c

```diff
@@ -366,6 +370,25 @@ static void s_s3_meta_request_default_request_finished(
         meta_request_default->synced_data.request_error_code = error_code;
 
         if (error_code == AWS_ERROR_SUCCESS) {
+            /* Send progress_callback for delivery on io_event_loop thread.
+             * For default meta-requests, we invoke the progress_callback once, after the sole HTTP request completes.
+             * This is simpler than reporting incremental progress as the response body is received,
+             * or the request body is streamed out, since then we'd also need to handle retries that reset
+             * progress back to 0% (our existing API only lets us report forward progress). */
+            if (meta_request->progress_callback != NULL) {
+                struct aws_s3_meta_request_event event = {.type = AWS_S3_META_REQUEST_EVENT_PROGRESS};
+                if (meta_request->type == AWS_S3_META_REQUEST_TYPE_PUT_OBJECT) {
```

Comment:
Debatable: check the request method to be PUT instead may makes more sense? For a PUT request, it's more likely to be interested in how much data was uploaded?

Reply:
There are lots of other S3 operations that use the "PUT" method.
For example: [CreateBucket](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateBucket.html#API_CreateBucket_RequestSyntax)

I looked at an old S3 [model.json](https://github.com/awslabs/aws-crt-java/blob/6a9012e7399185b6e4dc45372fcbe7df466078f6/s3-native-client/model/model.json) file to find this out

Reply:
Yeah, but even for those apis, will it make more sense to report how much was uploaded instead of received?